lib/ecto/query/builder.ex

```diff
@@ -404,6 +404,20 @@ defmodule Ecto.Query.Builder do
     {{:{}, [], [:over, [], [aggregate, window]]}, params_acc}
   end
 
+  def escape({:selected_as, _, [_expr, _name]}, _type, _params_acc, _vars, _env) do
```

Comment:
I forgot that escaping `selected_as/2` inside the select builder enforces it being at the root, because the only other escapes in there are for maps/tuples/structs, etc... everything else goes into this generic builder. So I put a nicer error message here to catch the non-root selects.

But I also formatted selected_as in the planner like you said (after collect_fields) to pick off stragglers (and also help with subqueries).